docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -0,0 +1,1946 @@
+.. Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Zero Downtime Smart Contract Upgrades
+#####################################
+
+.. .. toctree::
+   :hidden:
+
+Theory of Operation
+===================
+
+What are Zero Downtime Smart Contract Upgrades?
+-----------------------------------------------
+
+Zero-Downtime Smart Contract Upgrades (SCU) are a new feature for Daml
+packages which enable authors to publish new versions of their templates
+while maintaining compatibility with prior versions, without any
+downtime for package users and existing contracts.
+
+Previously, package authors could upgrade their packages using broadly
+one of two approaches:
+
+-  | Provide a Daml workflow for upgrading contracts to the new version,
+     and tell users of the old version to use the workflow to upgrade
+     their old templates to the new versions.
+   | This requires communication with all package users, splits package
+     users across the two versions during the migration, and may incur
+     significant network cost. This approach is described
+     :ref:`here <upgrade-overview>`.
+
+-  | Publish the new version to their participant, temporarily stop
+     workflows relating to the old version, and manually upgrade every
+     old template on the participant to the new version by directly
+     manipulating canton’s databases.
+   | This is error prone and requires some downtime for the participant.
+
+With SCU, any contract from the old package is automatically interpreted
+as the new version as soon as it is used in a new workflow that requires
+it.
+
+In this way, package authors can publish new package versions that
+automatically improve contracts’ functionality without requiring any
+error prone migrations, without downtime, without requiring any
+additional network traffic, and without any extensive communication with
+downstream users.
+
+Requirements
+------------
+
+Note that SCU is only available when the criteria below is met:
+
+-  Canton 2.9.x or above
+
+-  Daml LF Version 1.16 or above
+
+-  Canton Protocol Version 6 or above
+
+There are instructions below on how to configure this setup. The
+sections below, unless explicitly stated otherwise, assume that this is
+the case.
+
+How Does it Work
+----------------
+
+When upgrading a package, the package author modifies their existing
+package to add new functionality, such as new fields and choices. When
+the new package is uploaded to a participant with the old version, the
+participant ensures that it can automatically upgrade every change
```

Comment:
I've reworded this section, please let me know what you think

Reply:
"automatically upgrade every change" -- it's not "changes" that are upgraded is it? It is that "every changed contract is upgradeable" or "all changes between old and new version are backwards-compatible".

I'd suggest that you lean more on the notion of backwards-compatible changes, instead of "upgradeable changes".

Reply:
I think github hasn't shown you the reworded version, it's now

```
When
the new package is uploaded to a participant with the old version, 
the participant ensures that the every modification to the model in the
new version is a valid upgrade of the previous version.
```
I can reword the last bit to `in the new version is backwards-compatible with the previous version`, but I do like the phrasing of upgrade validity here, as it leads into the upgrade validity checks nicely.